libgraph/include/katana/analytics/ClusteringImplementationBase.h

```diff
@@ -489,16 +489,11 @@ struct ClusteringImplementationBase {
  */
   static katana::Result<std::unique_ptr<katana::PropertyGraph>>
   DuplicateGraphWithSameTopo(const katana::PropertyGraph& pfg_from) {
-    const katana::GraphTopology& topology_from = pfg_from.topology();
-
-    katana::GraphTopology topo_copy = GraphTopology::Copy(topology_from);
-
-    auto pfg_to_res = katana::PropertyGraph::Make(std::move(topo_copy));
-    if (!pfg_to_res) {
-      return pfg_to_res.error();
-    }
-    return std::unique_ptr<katana::PropertyGraph>(
-        std::move(pfg_to_res.value()));
+    // The resulting PropertyGraph is detached from the property table of the input graph.
+    // Thus any property index indirection of the original topology has to be dropped.
+    katana::GraphTopology topo_copy =
+        GraphTopology::CopyWithoutPropertyIndexes(pfg_from.topology());
+    return katana::PropertyGraph::Make(std::move(topo_copy));
```

Comment:
what will happen to the underlying edge weights in the copied graph?

Reply:
This function creates a brand-new property graph that contains a copy of the topology and nothing else. There are no edge weights. I didn't change any of that logic. What this PR does is that it makes sure that when a new property is created on this new property graph, we can index into the property column correctly, by dropping the original projected-to-original mappings.

Reply:
then either this function is not being used anymore or something else is going on. I will check this then. thanks

Reply:
ok, it seems that edge property is calculated in a separate function, so it seems fine. thanks